### PR TITLE
[PIR] Fix `check_view_api_used_by_inplace`

### PR DESCRIPTION
**Pull request by ooooo-create**
### PR Category

Execute Infrastructure

### PR Types

Others

### Description

https://github.com/PaddlePaddle/PaddleDetection/blob/0914611178e9db34e746a351c7fc090048eeebc0/ppdet/modeling/losses/ssd_loss.py#L115

- 修复检查逻辑， value.detach 使用的 `pd_op.share_data_` 不会和其他 inplace api 一样立刻修改显存
- 但是产生的新的 Value 在逻辑上是不保证安全的，因为之后可能会被其他 inplace api 使用，也修改了之前的显存，目前的检查没有包含这种情况（现在只是跳过了

```python
z = paddle.squeeze(x,-1)
y = x.detach() #不会立刻导致 z 的显存变化
# y.add_(xxx) ，就会导致 z 的显存变化

```

### CODE REVIEW

File: python/paddle/jit/dy2static/program_translator.py [hunk 1]
```diff
@@ -120,24 +120,31 @@ def check_view_api_used_by_inplace(program: paddle.pir.Program) -> None:
...
-                op.operand_source(index).is_same(value)
-                for index in inplace_info.keys()
-            ):
+            if is_used_by_inplace_op(op, value, inplace_info):
+                # TODO(ooooo): Deal with these inplace ops
                 if (
                     op.name() == "pd_op.set_value_"
                     or op.name() == "pd_op.set_value_with_tensor_"
+                    # It willn't change tensor imdeiately,but it's ouput is dangerous.
+                    or op.name() == "pd_op.share_data_"
```
Comment (SigureMo) [inline python/paddle/jit/dy2static/program_translator.py]: op.name() in ["a", "b", "c"]

Comment (ooooo-create) [reply python/paddle/jit/dy2static/program_translator.py]: 已修改

**Review by SigureMo — APPROVED** [review]
LGTMeow
